tests/libnet/job/job.go

```diff
@@ -139,10 +138,14 @@ func WaitForJob(job *batchv1.Job, toSucceed bool, timeout time.Duration) error {
 				break
 			case batchv1.JobFailureTarget:
 				break
+			case batchv1.JobSuccessCriteriaMet:
+				break
 			}
 		}
 		return !finish, nil
-	})
+	}
+
+	err := wait.PollUntilContextTimeout(context.Background(), time.Second, timeout, true, poller)
```

Comment:
Please use:
`Expect(err).ToNot(HaveOccurred(),fmt.Sprintf("job %s timeout reached, status: %+v, err: %v", job.Name, job.Status, err))`

Reply:
According to [writing-good-e2e-tests manual](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-testing/writing-good-e2e-tests.md#interrupting-tests):
```
Using gomega assertions in helper packages is problematic for two reasons:

The stacktrace associated with the failure starts with the helper unless extra care is take to pass in a stack offset.
Additional explanations for a potential failure must be prepared beforehand and passed in.
```
So I think it's better to return the error. Do you agree?

Reply:
Yes, you are right. Sorry, I thought it was a test. :)